cs-xml/src/main/java/io/cloudslang/content/xml/actions/ApplyXslTransformation.java

```diff
@@ -0,0 +1,81 @@
+     * @param outputFile      The local file to write the output of the transformation. If an output file is not specified
+     *                        the output of the transformation will be returned as returnResult.
+     * @param parsingFeatures The list of XML parsing features separated by new line (CRLF).
+     *                        The feature name - value must be separated by empty space.
+     *                        Setting specific features this field could be used to avoid XML security issues like
+     *                        "XML Entity Expansion injection" and "XML External Entity injection".
+     *                        To avoid aforementioned security issues we strongly recommend to set this input to the following values:
+     *                        http://apache.org/xml/features/disallow-doctype-decl true
+     *                        http://xml.org/sax/features/external-general-entities false
+     *                        http://xml.org/sax/features/external-parameter-entities false
+     *                        When the "http://apache.org/xml/features/disallow-doctype-decl" feature is set to "true"
+     *                        the parser will throw a FATAL ERROR if the incoming document contains a DOCTYPE declaration.
+     *                        When the "http://xml.org/sax/features/external-general-entities" feature is set to "false"
+     *                        the parser will not include external general entities.
+     *                        When the "http://xml.org/sax/features/external-parameter-entities" feature is set to "false"
+     *                        the parser will not include external parameter entities or the external DTD subset.
+     *                        If any of the validations fails, the operation will fail with an error message describing the problem.
+     *                        Default value:
+     *                        http://apache.org/xml/features/disallow-doctype-decl true
+     *                        http://xml.org/sax/features/external-general-entities false
+     *                        http://xml.org/sax/features/external-parameter-entities false
+     * @return The output of the transformation, if no output file is specified.
+     */
+    @Action(name = "Apply XSL Transformation",
+            outputs = {
+                    @Output(RETURN_CODE),
+                    @Output(RETURN_RESULT),
+                    @Output(EXCEPTION)},
+            responses = {
+                    @Response(text = ResponseNames.SUCCESS, field = RETURN_CODE, value = ReturnCodes.SUCCESS, matchType = MatchType.COMPARE_EQUAL),
+                    @Response(text = ResponseNames.FAILURE, field = RETURN_CODE, value = ReturnCodes.FAILURE, matchType = MatchType.COMPARE_EQUAL, isDefault = true, isOnFail = true)})
+    public Map<String, String> applyXslTransformation(
+            @Param(value = Constants.Inputs.XML_DOCUMENT) String xmlDocument,
```

Comment:
Yes it is, if xml document is not supplied, but an XSL Template is, the resulting HTML file will transform the template only. The transformation can be done with our without an xml document.